Linting python code

Linting tools can help catch errors and ensure a consistent style. For python, flake8 is a good bundle. Right now, it produces a couple hundred warnings:

<details>

```
hubmap$ flake8
./frontend/models.py:1:1: F401 'django.db.models' imported but unused
./frontend/admin.py:1:1: F401 'django.contrib.admin' imported but unused
./frontend/tests.py:1:1: F401 'django.test.TestCase' imported but unused
./frontend/views.py:2:1: F401 'django.http.HttpResponse' imported but unused
./frontend/views.py:3:1: F401 'django.conf.settings' imported but unused
./frontend/views.py:4:1: F401 'os' imported but unused
./frontend/views.py:5:1: F401 'logging' imported but unused
./frontend/views.py:13:1: E302 expected 2 blank lines, found 1
./frontend/views.py:16:1: E302 expected 2 blank lines, found 1
./frontend/views.py:41:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:3:1: F401 'django.db.models.QuerySet' imported but unused
./dataportal/models.py:5:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:24:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:28:80: E501 line too long (111 > 79 characters)
./dataportal/models.py:35:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:43:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:49:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:55:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:61:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:89:1: E305 expected 2 blank lines after class or function definition, found 1
./dataportal/models.py:90:1: E302 expected 2 blank lines, found 0
./dataportal/models.py:92:80: E501 line too long (80 > 79 characters)
./dataportal/models.py:98:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:102:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:110:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:115:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:119:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:124:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:128:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:130:80: E501 line too long (106 > 79 characters)
./dataportal/models.py:132:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:140:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:144:1: E302 expected 2 blank lines, found 1
./dataportal/models.py:148:1: W391 blank line at end of file
./dataportal/serializers.py:2:1: F401 'rest_framework.serializers.ListSerializer' imported but unused
./dataportal/serializers.py:4:1: F403 'from .models import *' used; unable to detect undefined names
./dataportal/serializers.py:5:1: E265 block comment should start with '# '
./dataportal/serializers.py:7:1: E265 block comment should start with '# '
./dataportal/serializers.py:9:1: E302 expected 2 blank lines, found 1
./dataportal/serializers.py:19:17: F405 'Institution' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:21:1: E302 expected 2 blank lines, found 1
./dataportal/serializers.py:24:9: E122 continuation line missing indentation or outdented
./dataportal/serializers.py:25:9: E122 continuation line missing indentation or outdented
./dataportal/serializers.py:26:9: E122 continuation line missing indentation or outdented
./dataportal/serializers.py:27:9: E122 continuation line missing indentation or outdented
./dataportal/serializers.py:28:9: E122 continuation line missing indentation or outdented
./dataportal/serializers.py:29:9: E122 continuation line missing indentation or outdented
./dataportal/serializers.py:30:9: E122 continuation line missing indentation or outdented
./dataportal/serializers.py:31:9: E122 continuation line missing indentation or outdented
./dataportal/serializers.py:32:9: E122 continuation line missing indentation or outdented
./dataportal/serializers.py:33:9: E122 continuation line missing indentation or outdented
./dataportal/serializers.py:47:17: F405 'TissueExpressionHeatmap' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:49:1: E302 expected 2 blank lines, found 1
./dataportal/serializers.py:59:17: F405 'Tissue' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:61:1: E302 expected 2 blank lines, found 1
./dataportal/serializers.py:71:17: F405 'DataType' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:73:1: E302 expected 2 blank lines, found 1
./dataportal/serializers.py:90:9: E128 continuation line under-indented for visual indent
./dataportal/serializers.py:90:10: E201 whitespace after '{'
./dataportal/serializers.py:99:27: E231 missing whitespace after ','
./dataportal/serializers.py:100:17: F405 'Gene' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:103:1: E302 expected 2 blank lines, found 1
./dataportal/serializers.py:117:37: E203 whitespace before ':'
./dataportal/serializers.py:117:80: E501 line too long (100 > 79 characters)
./dataportal/serializers.py:118:17: F405 'Protein' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:121:1: E302 expected 2 blank lines, found 1
./dataportal/serializers.py:141:17: F405 'Study' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:146:35: F405 'ScRnaSeqStudyCDNA' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:147:80: E501 line too long (96 > 79 characters)
./dataportal/serializers.py:149:35: F405 'ScRnaSeqStudyBarcoded' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:150:80: E501 line too long (100 > 79 characters)
./dataportal/serializers.py:152:17: E203 whitespace before ':'
./dataportal/serializers.py:153:35: F405 'Study' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:154:26: F405 'Study' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:170:17: F405 'ScRnaSeqStudyCDNA' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:174:23: F405 'ScRnaSeqStudyCDNA' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:177:1: E302 expected 2 blank lines, found 1
./dataportal/serializers.py:180:5: E301 expected 1 blank line, found 0
./dataportal/serializers.py:191:37: E203 whitespace before ':'
./dataportal/serializers.py:191:80: E501 line too long (100 > 79 characters)
./dataportal/serializers.py:192:69: E222 multiple spaces after operator
./dataportal/serializers.py:196:17: F405 'ScRnaSeqStudyBarcoded' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:199:30: F405 'ScRnaSeqStudyBarcoded' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:199:80: E501 line too long (83 > 79 characters)
./dataportal/serializers.py:203:1: E302 expected 2 blank lines, found 1
./dataportal/serializers.py:205:5: E301 expected 1 blank line, found 0
./dataportal/serializers.py:214:17: F405 'ScAtacSeqStudy' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:217:19: F405 'ScAtacSeqStudy' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:221:1: E302 expected 2 blank lines, found 1
./dataportal/serializers.py:223:5: E301 expected 1 blank line, found 0
./dataportal/serializers.py:224:9: E265 block comment should start with '# '
./dataportal/serializers.py:228:37: E203 whitespace before ':'
./dataportal/serializers.py:228:80: E501 line too long (100 > 79 characters)
./dataportal/serializers.py:229:67: E225 missing whitespace around operator
./dataportal/serializers.py:233:17: F405 'SpatialTranscriptomicStudy' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:236:33: F405 'SpatialTranscriptomicStudy' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:236:80: E501 line too long (91 > 79 characters)
./dataportal/serializers.py:240:1: E302 expected 2 blank lines, found 1
./dataportal/serializers.py:242:5: E301 expected 1 blank line, found 0
./dataportal/serializers.py:250:80: E501 line too long (88 > 79 characters)
./dataportal/serializers.py:251:40: E203 whitespace before ':'
./dataportal/serializers.py:251:65: E231 missing whitespace after ','
./dataportal/serializers.py:251:72: E231 missing whitespace after ','
./dataportal/serializers.py:251:80: E501 line too long (91 > 79 characters)
./dataportal/serializers.py:251:81: E231 missing whitespace after ','
./dataportal/serializers.py:251:88: E231 missing whitespace after ','
./dataportal/serializers.py:252:17: F405 'MassCytometryStudy' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:255:32: F405 'MassCytometryStudy' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:255:80: E501 line too long (82 > 79 characters)
./dataportal/serializers.py:259:1: E302 expected 2 blank lines, found 1
./dataportal/serializers.py:268:80: E501 line too long (88 > 79 characters)
./dataportal/serializers.py:269:17: F405 'MicroscopyStudy' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:272:22: F405 'MicroscopyStudy' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:276:1: E302 expected 2 blank lines, found 1
./dataportal/serializers.py:285:80: E501 line too long (88 > 79 characters)
./dataportal/serializers.py:286:17: F405 'SeqFishImagingStudy' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:289:28: F405 'SeqFishImagingStudy' may be undefined, or defined from star imports: .models
./dataportal/serializers.py:293:1: E302 expected 2 blank lines, found 1
./dataportal/serializers.py:298:1: E302 expected 2 blank lines, found 1
./dataportal/serializers.py:305:21: E225 missing whitespace around operator
./dataportal/serializers.py:308:17: F405 'Study' may be undefined, or defined from star imports: .models
./dataportal/admin.py:17:36: W292 no newline at end of file
./dataportal/utils.py:4:1: F403 'from .models import *' used; unable to detect undefined names
./dataportal/utils.py:5:1: F403 'from .serializers import *' used; unable to detect undefined names
./dataportal/utils.py:7:80: E501 line too long (106 > 79 characters)
./dataportal/utils.py:8:1: E302 expected 2 blank lines, found 0
./dataportal/utils.py:8:27: F405 'serializers' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:13:1: E302 expected 2 blank lines, found 1
./dataportal/utils.py:15:31: F405 'ScRnaSeqStudyCDNA' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:16:16: F405 'ScRnaSeqStudyCDNASerializer' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:17:33: F405 'ScRnaSeqStudyBarcoded' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:18:16: F405 'ScRnaSeqStudyBarcodedSerializer' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:19:33: F405 'SpatialTranscriptomicStudy' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:20:16: F405 'SpatialTranscriptomicStudySerializer' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:21:33: F405 'MassCytometryStudy' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:22:16: F405 'MassCytometryStudySerializer' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:23:33: F405 'MicroscopyStudy' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:24:16: F405 'MicroscopyStudySerializer' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:25:33: F405 'SeqFishImagingStudy' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:26:16: F405 'SeqFishImagingStudySerializer' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:28:1: E302 expected 2 blank lines, found 1
./dataportal/utils.py:30:5: E265 block comment should start with '# '
./dataportal/utils.py:38:8: E714 test for object identity should be 'is not'
./dataportal/utils.py:42:80: E501 line too long (117 > 79 characters)
./dataportal/utils.py:45:9: E265 block comment should start with '# '
./dataportal/utils.py:47:80: E501 line too long (111 > 79 characters)
./dataportal/utils.py:49:5: E265 block comment should start with '# '
./dataportal/utils.py:53:5: E265 block comment should start with '# '
./dataportal/utils.py:54:34: F405 'ScRnaSeqStudyBarcodedSerializer' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:54:80: E501 line too long (109 > 79 characters)
./dataportal/utils.py:55:29: F405 'ScRnaSeqStudyCDNASerializer' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:55:80: E501 line too long (95 > 79 characters)
./dataportal/utils.py:56:41: F405 'SpatialTranscriptomicStudySerializer' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:56:80: E501 line too long (111 > 79 characters)
./dataportal/utils.py:57:80: E501 line too long (93 > 79 characters)
./dataportal/utils.py:58:29: F405 'MicroscopyStudySerializer' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:58:80: E501 line too long (93 > 79 characters)
./dataportal/utils.py:59:33: F405 'MassCytometryStudySerializer' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:59:80: E501 line too long (104 > 79 characters)
./dataportal/utils.py:60:29: F405 'ScAtacSeqStudySerializer' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:60:80: E501 line too long (92 > 79 characters)
./dataportal/utils.py:61:35: F405 'SeqFishImagingStudySerializer' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:61:80: E501 line too long (109 > 79 characters)
./dataportal/utils.py:62:80: E501 line too long (108 > 79 characters)
./dataportal/utils.py:63:13: E127 continuation line over-indented for visual indent
./dataportal/utils.py:63:80: E501 line too long (114 > 79 characters)
./dataportal/utils.py:64:16: E127 continuation line over-indented for visual indent
./dataportal/utils.py:66:8: E714 test for object identity should be 'is not'
./dataportal/utils.py:67:28: F405 'GeneSerializer' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:68:31: F405 'ProteinSerializer' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:68:80: E501 line too long (85 > 79 characters)
./dataportal/utils.py:72:1: E302 expected 2 blank lines, found 1
./dataportal/utils.py:74:26: F405 'MassCytometryStudy' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:76:26: F405 'MassCytometryStudy' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:77:80: E501 line too long (84 > 79 characters)
./dataportal/utils.py:78:80: E501 line too long (85 > 79 characters)
./dataportal/utils.py:79:80: E501 line too long (84 > 79 characters)
./dataportal/utils.py:80:9: E122 continuation line missing indentation or outdented
./dataportal/utils.py:83:1: E302 expected 2 blank lines, found 1
./dataportal/utils.py:85:27: F405 'ScRnaSeqStudyBarcoded' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:87:27: F405 'ScRnaSeqStudyBarcoded' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:88:80: E501 line too long (84 > 79 characters)
./dataportal/utils.py:89:80: E501 line too long (85 > 79 characters)
./dataportal/utils.py:90:80: E501 line too long (86 > 79 characters)
./dataportal/utils.py:93:1: E302 expected 2 blank lines, found 1
./dataportal/utils.py:95:22: F405 'ScRnaSeqStudyCDNA' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:97:22: F405 'ScRnaSeqStudyCDNA' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:98:80: E501 line too long (84 > 79 characters)
./dataportal/utils.py:99:80: E501 line too long (85 > 79 characters)
./dataportal/utils.py:100:80: E501 line too long (86 > 79 characters)
./dataportal/utils.py:103:1: E302 expected 2 blank lines, found 1
./dataportal/utils.py:105:22: F405 'ScAtacSeqStudy' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:107:22: F405 'ScAtacSeqStudy' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:108:80: E501 line too long (84 > 79 characters)
./dataportal/utils.py:109:80: E501 line too long (85 > 79 characters)
./dataportal/utils.py:110:80: E501 line too long (84 > 79 characters)
./dataportal/utils.py:113:1: E302 expected 2 blank lines, found 1
./dataportal/utils.py:115:34: F405 'SpatialTranscriptomicStudy' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:117:34: F405 'SpatialTranscriptomicStudy' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:118:80: E501 line too long (84 > 79 characters)
./dataportal/utils.py:119:80: E501 line too long (85 > 79 characters)
./dataportal/utils.py:123:1: E302 expected 2 blank lines, found 1
./dataportal/utils.py:125:22: F405 'MicroscopyStudy' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:127:22: F405 'MicroscopyStudy' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:127:80: E501 line too long (124 > 79 characters)
./dataportal/utils.py:128:22: E128 continuation line under-indented for visual indent
./dataportal/utils.py:128:80: E501 line too long (127 > 79 characters)
./dataportal/utils.py:128:95: E225 missing whitespace around operator
./dataportal/utils.py:132:1: E302 expected 2 blank lines, found 1
./dataportal/utils.py:134:17: F405 'Gene' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:136:17: F405 'Gene' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:139:1: E302 expected 2 blank lines, found 1
./dataportal/utils.py:144:5: E265 block comment should start with '# '
./dataportal/utils.py:149:11: F405 'GeneSerializer' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:150:5: E265 block comment should start with '# '
./dataportal/utils.py:151:16: F405 'GeneSerializer' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:154:1: E302 expected 2 blank lines, found 1
./dataportal/utils.py:156:20: F405 'Protein' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:158:20: F405 'Protein' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:158:80: E501 line too long (107 > 79 characters)
./dataportal/utils.py:161:1: E302 expected 2 blank lines, found 1
./dataportal/utils.py:163:28: F405 'SeqFishImagingStudy' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:165:28: F405 'SeqFishImagingStudy' may be undefined, or defined from star imports: .models, .serializers
./dataportal/utils.py:165:80: E501 line too long (134 > 79 characters)
./dataportal/utils.py:166:22: E128 continuation line under-indented for visual indent
./dataportal/utils.py:166:80: E501 line too long (127 > 79 characters)
./dataportal/utils.py:166:95: E225 missing whitespace around operator
./dataportal/utils.py:170:1: E302 expected 2 blank lines, found 1
./dataportal/utils.py:178:1: W391 blank line at end of file
./dataportal/tests.py:4:1: F403 'from .models import *' used; unable to detect undefined names
./dataportal/tests.py:8:1: E302 expected 2 blank lines, found 1
./dataportal/tests.py:12:14: F405 'DataType' may be undefined, or defined from star imports: .models
./dataportal/tests.py:13:13: F405 'Tissue' may be undefined, or defined from star imports: .models
./dataportal/tests.py:14:13: F405 'Institution' may be undefined, or defined from star imports: .models
./dataportal/tests.py:15:9: F405 'ScRnaSeqStudyCDNA' may be undefined, or defined from star imports: .models
./dataportal/tests.py:26:17: F405 'Study' may be undefined, or defined from star imports: .models
./dataportal/tests.py:31:17: F405 'Study' may be undefined, or defined from star imports: .models
./dataportal/tests.py:50:80: E501 line too long (80 > 79 characters)
./dataportal/urls.py:2:1: F403 'from django.conf.urls import *' used; unable to detect undefined names
./dataportal/urls.py:10:5: F405 'url' may be undefined, or defined from star imports: django.conf.urls
./dataportal/urls.py:11:5: F405 'url' may be undefined, or defined from star imports: django.conf.urls
./dataportal/urls.py:12:5: F405 'url' may be undefined, or defined from star imports: django.conf.urls
./dataportal/views.py:1:1: F401 'collections.defaultdict' imported but unused
./dataportal/views.py:7:1: F403 'from .utils import *' used; unable to detect undefined names
./dataportal/views.py:8:1: F403 'from .serializers import *' used; unable to detect undefined names
./dataportal/views.py:12:1: F401 'json.loads' imported but unused
./dataportal/views.py:12:1: F401 'json.dumps' imported but unused
./dataportal/views.py:14:1: F401 'django.views.generic.View' imported but unused
./dataportal/views.py:15:1: F401 'django.http.HttpResponse' imported but unused
./dataportal/views.py:16:1: F401 'django.conf.settings' imported but unused
./dataportal/views.py:17:1: F401 'os' imported but unused
./dataportal/views.py:18:1: F401 'logging' imported but unused
./dataportal/views.py:20:1: E265 block comment should start with '# '
./dataportal/views.py:21:1: E265 block comment should start with '# '
./dataportal/views.py:23:1: E302 expected 2 blank lines, found 1
./dataportal/views.py:24:16: E222 multiple spaces after operator
./dataportal/views.py:27:1: E302 expected 2 blank lines, found 1
./dataportal/views.py:28:24: F405 'StudyListSerializer' may be undefined, or defined from star imports: .serializers, .utils
./dataportal/views.py:31:20: F405 'get_response_for_request' may be undefined, or defined from star imports: .serializers, .utils
./dataportal/views.py:32:80: E501 line too long (110 > 79 characters)
./dataportal/views.py:33:80: E501 line too long (112 > 79 characters)
./dataportal/views.py:34:35: E203 whitespace before ':'
./dataportal/views.py:34:40: E201 whitespace after '{'
./dataportal/views.py:34:53: E203 whitespace before ':'
./dataportal/views.py:34:75: E231 missing whitespace after ','
./dataportal/views.py:34:80: E501 line too long (114 > 79 characters)
./dataportal/views.py:41:80: E501 line too long (87 > 79 characters)
./dataportal/views.py:45:1: E265 block comment should start with '# '
./dataportal/views.py:47:22: F405 'StudySerializer' may be undefined, or defined from star imports: .serializers, .utils
./dataportal/views.py:53:1: E302 expected 2 blank lines, found 1
./dataportal/views.py:54:24: F405 'StudyListSerializer' may be undefined, or defined from star imports: .serializers, .utils
./dataportal/views.py:58:20: F405 'get_response_for_request' may be undefined, or defined from star imports: .serializers, .utils
./dataportal/views.py:63:1: E302 expected 2 blank lines, found 1
./dataportal/views.py:64:24: F405 'GeneSerializer' may be undefined, or defined from star imports: .serializers, .utils
./dataportal/views.py:67:20: F405 'get_genes' may be undefined, or defined from star imports: .serializers, .utils
./dataportal/views.py:70:1: E302 expected 2 blank lines, found 1
./dataportal/views.py:71:24: F405 'StudyListSerializer' may be undefined, or defined from star imports: .serializers, .utils
./dataportal/views.py:74:20: F405 'get_response_for_request' may be undefined, or defined from star imports: .serializers, .utils
./dataportal/views.py:77:1: E302 expected 2 blank lines, found 1
./dataportal/views.py:78:16: F405 'Study' may be undefined, or defined from star imports: .serializers, .utils
./dataportal/views.py:81:16: F405 'get_serializer_class' may be undefined, or defined from star imports: .serializers, .utils
./dataportal/views.py:84:80: E501 line too long (81 > 79 characters)
./dataportal/views.py:87:1: E302 expected 2 blank lines, found 1
./dataportal/views.py:90:1: E302 expected 2 blank lines, found 1
./dataportal/views.py:109:39: E203 whitespace before ':'
./dataportal/views.py:109:80: E501 line too long (92 > 79 characters)
./dataportal/views.py:113:19: F405 'TissueColorSerializer' may be undefined, or defined from star imports: .serializers, .utils
./dataportal/views.py:123:1: E305 expected 2 blank lines after class or function definition, found 1
./dataportal/views.py:124:16: F405 'Tissue' may be undefined, or defined from star imports: .serializers, .utils
./dataportal/views.py:125:21: F405 'Institution' may be undefined, or defined from star imports: .serializers, .utils
./dataportal/views.py:126:19: F405 'DataType' may be undefined, or defined from star imports: .serializers, .utils
./dataportal/views.py:129:1: E302 expected 2 blank lines, found 1
./dataportal/views.py:152:80: E501 line too long (92 > 79 characters)
./dataportal/views.py:157:1: E302 expected 2 blank lines, found 1
./dataportal/views.py:162:80: E501 line too long (86 > 79 characters)
./dataportal/views.py:167:80: E501 line too long (92 > 79 characters)
./dataportal/migrations/0001_initial.py:19:80: E501 line too long (114 > 79 characters)
./dataportal/migrations/0001_initial.py:26:80: E501 line too long (114 > 79 characters)
./dataportal/migrations/0001_initial.py:27:80: E501 line too long (86 > 79 characters)
./dataportal/migrations/0001_initial.py:28:80: E501 line too long (88 > 79 characters)
./dataportal/migrations/0001_initial.py:29:80: E501 line too long (87 > 79 characters)
./dataportal/migrations/0001_initial.py:35:80: E501 line too long (114 > 79 characters)
./dataportal/migrations/0001_initial.py:42:80: E501 line too long (114 > 79 characters)
./dataportal/migrations/0001_initial.py:44:80: E501 line too long (120 > 79 characters)
./dataportal/migrations/0001_initial.py:45:80: E501 line too long (125 > 79 characters)
./dataportal/migrations/0001_initial.py:46:80: E501 line too long (124 > 79 characters)
./dataportal/migrations/0001_initial.py:55:80: E501 line too long (114 > 79 characters)
./dataportal/migrations/0001_initial.py:62:80: E501 line too long (114 > 79 characters)
./dataportal/migrations/0001_initial.py:78:80: E501 line too long (192 > 79 characters)
./dataportal/migrations/0001_initial.py:80:80: E501 line too long (122 > 79 characters)
./dataportal/migrations/0001_initial.py:87:80: E501 line too long (192 > 79 characters)
./dataportal/migrations/0001_initial.py:96:80: E501 line too long (192 > 79 characters)
./dataportal/migrations/0001_initial.py:105:80: E501 line too long (192 > 79 characters)
./dataportal/migrations/0001_initial.py:117:80: E501 line too long (105 > 79 characters)
./dataportal/migrations/0001_initial.py:122:80: E501 line too long (114 > 79 characters)
./dataportal/migrations/0001_initial.py:123:80: E501 line too long (82 > 79 characters)
./dataportal/migrations/0001_initial.py:124:80: E501 line too long (83 > 79 characters)
./dataportal/migrations/0001_initial.py:125:80: E501 line too long (111 > 79 characters)
./dataportal/migrations/0001_initial.py:131:80: E501 line too long (133 > 79 characters)
./dataportal/migrations/0001_initial.py:136:80: E501 line too long (206 > 79 characters)
./dataportal/migrations/0001_initial.py:146:80: E501 line too long (208 > 79 characters)
./dataportal/migrations/0001_initial.py:154:80: E501 line too long (206 > 79 characters)
./dataportal/migrations/0001_initial.py:161:80: E501 line too long (192 > 79 characters)
./dataportal/migrations/0001_initial.py:169:80: E501 line too long (208 > 79 characters)
./dataportal/migrations/0001_initial.py:178:80: E501 line too long (206 > 79 characters)
./dataportal/management/commands/studies2csv.py:5:21: E201 whitespace after '['
./dataportal/management/commands/studies2csv.py:5:58: E231 missing whitespace after ','
./dataportal/management/commands/studies2csv.py:5:77: E231 missing whitespace after ','
./dataportal/management/commands/studies2csv.py:5:80: E501 line too long (90 > 79 characters)
./dataportal/management/commands/studies2csv.py:6:23: E127 continuation line over-indented for visual indent
./dataportal/management/commands/studies2csv.py:6:80: E501 line too long (89 > 79 characters)
./dataportal/management/commands/studies2csv.py:14:80: E501 line too long (86 > 79 characters)
./dataportal/management/commands/studies2csv.py:15:14: F821 undefined name 'csv'
./dataportal/management/commands/studies2csv.py:19:9: F841 local variable 'row' is assigned to but never used
./dataportal/management/commands/studies2csv.py:19:32: F821 undefined name 'get_attribute'
./dataportal/management/commands/studies2csv.py:19:80: E501 line too long (85 > 79 characters)
./dataportal/management/commands/studies2csv.py:21:5: E265 block comment should start with '# '
./dataportal/management/commands/studies2csv.py:22:20: W292 no newline at end of file
./dataportal/management/commands/csv2models.py:3:1: F403 'from dataportal.models import *' used; unable to detect undefined names
./dataportal/management/commands/csv2models.py:5:1: F811 redefinition of unused 'csv' from line 1
./dataportal/management/commands/csv2models.py:6:1: F401 'django.core.management.base.CommandError' imported but unused
./dataportal/management/commands/csv2models.py:7:1: F401 'sys' imported but unused
./dataportal/management/commands/csv2models.py:8:1: F401 'dataportal.models.Study' imported but unused
./dataportal/management/commands/csv2models.py:13:9: E225 missing whitespace around operator
./dataportal/management/commands/csv2models.py:14:5: E301 expected 1 blank line, found 0
./dataportal/management/commands/csv2models.py:17:9: F841 local variable 'model' is assigned to but never used
./dataportal/management/commands/csv2models.py:17:41: F405 'model_name' may be undefined, or defined from star imports: dataportal.models
./dataportal/management/commands/csv2models.py:18:12: F405 'model_name' may be undefined, or defined from star imports: dataportal.models
./dataportal/management/commands/csv2models.py:24:21: F841 local variable 'p' is assigned to but never used
./dataportal/management/commands/csv2models.py:25:1: W391 blank line at end of file
./hubmap/production_settings.py:7:1: E402 module level import not at top of file
./hubmap/settings.py:11:80: E501 line too long (87 > 79 characters)
./hubmap/settings.py:30:80: E501 line too long (88 > 79 characters)
./hubmap/settings.py:64:80: E501 line too long (86 > 79 characters)
./hubmap/settings.py:92:80: E501 line too long (91 > 79 characters)
./hubmap/settings.py:95:80: E501 line too long (81 > 79 characters)
./hubmap/settings.py:98:80: E501 line too long (82 > 79 characters)
./hubmap/settings.py:101:80: E501 line too long (83 > 79 characters)
./hubmap/settings.py:123:80: E501 line too long (86 > 79 characters)
./hubmap/settings.py:130:1: E302 expected 2 blank lines, found 0
./hubmap/settings.py:132:80: E501 line too long (81 > 79 characters)
./hubmap/settings.py:162:1: E305 expected 2 blank lines after class or function definition, found 1
./hubmap/settings.py:166:5: F403 'from .local_settings import *' used; unable to detect undefined names
./hubmap/settings.py:166:5: F401 '.local_settings.*' imported but unused
./fixtures/add_tissuegene_heatmaps2db.py:2:1: F401 'numpy as np' imported but unused
./fixtures/add_tissuegene_heatmaps2db.py:3:1: F401 'pandas as pd' imported but unused
./fixtures/add_tissuegene_heatmaps2db.py:4:1: F403 'from dataportal.models import *' used; unable to detect undefined names
./fixtures/add_tissuegene_heatmaps2db.py:6:1: E302 expected 2 blank lines, found 1
./fixtures/add_tissuegene_heatmaps2db.py:9:1: E302 expected 2 blank lines, found 1
./fixtures/add_tissuegene_heatmaps2db.py:14:11: F405 'TissueExpressionHeatmap' may be undefined, or defined from star imports: dataportal.models
./fixtures/add_tissuegene_heatmaps2db.py:17:15: F405 'Gene' may be undefined, or defined from star imports: dataportal.models
./fixtures/add_tissuegene_heatmaps2db.py:23:1: E305 expected 2 blank lines after class or function definition, found 1
```

</details>

I would propose making an initial PR that brings in flake8, and ignores all the errors, and if we're agreed that this is a reasonable standard to adhere to, making successive PRs to fix the warnings.

--- a/hubmap/dataportal/models.py
+++ b/hubmap/dataportal/models.py
@@ -26,7 +26,8 @@ class Gene(models.Model):
     entrez_id = models.CharField(max_length=50, blank=True, null=True)
     hugo_symbol = models.CharField(max_length=50, blank=True, null=True)
     ensembl_id = models.CharField(max_length=50, blank=True, null=True)
-    tissue_expression_heatmap = models.ForeignKey(TissueExpressionHeatmap, on_delete=models.CASCADE, null=True)
+    tissue_expression_heatmap = models.ForeignKey(
+        TissueExpressionHeatmap, on_delete=models.CASCADE, null=True)
 
     # TODO: any other
 
@@ -143,7 +144,8 @@ class SpatialTranscriptomicStudy(Study):
 
 class ImagingStudy(Study):
     image_count = models.PositiveIntegerField()
-    preview_image = models.ImageField(max_length=500, upload_to='gallery/%Y/%m/%d', null=True, blank=True)
+    preview_image = models.ImageField(
+        max_length=500, upload_to='gallery/%Y/%m/%d', null=True, blank=True)
 
 
 @study_type

--- a/hubmap/dataportal/serializers.py
+++ b/hubmap/dataportal/serializers.py
@@ -1,6 +1,21 @@
 from rest_framework import serializers
 
-from .models import *
+from .models import (
+    DataType,
+    Gene,
+    Institution,
+    MassCytometryStudy,
+    MicroscopyStudy,
+    Protein,
+    ScAtacSeqStudy,
+    ScRnaSeqStudyBarcoded,
+    ScRnaSeqStudyCDNA,
+    SeqFishImagingStudy,
+    SpatialTranscriptomicStudy,
+    Study,
+    Tissue,
+    TissueExpressionHeatmap
+)
 
 #TODO : add create and update and delete, put for all serializers
 # (all update/add/delete requests for scripts to write metadata to models)

--- a/hubmap/dataportal/serializers.py
+++ b/hubmap/dataportal/serializers.py
@@ -17,9 +17,9 @@ from .models import (
     TissueExpressionHeatmap
 )
 
-#TODO : add create and update and delete, put for all serializers
+# TODO : add create and update and delete, put for all serializers
 # (all update/add/delete requests for scripts to write metadata to models)
-#TODO: add tests for django restframework
+# TODO: add tests for django restframework
 
 
 class InstitutionSerializer(serializers.ModelSerializer):
@@ -250,7 +250,6 @@ class SpatialTranscriptomicStudySerializer(StudySerializer):
     genes = GeneSerializer(many=True, read_only=True)
 
     class Meta(StudySerializer.Meta):
-        #list_serializer_class = StudyListSerializer
         fields = StudySerializer.Meta.fields + (
             'genes',
         )

--- a/hubmap/dataportal/serializers.py
+++ b/hubmap/dataportal/serializers.py
@@ -154,7 +154,8 @@ class StudySerializer(serializers.ModelSerializer):
                 return ScRnaSeqStudyCDNASerializer(context=self.context).to_internal_value(data)
             elif data.get('subclass') == "ScRnaSeqStudyCDNASerializer":
                 self.Meta.model = ScRnaSeqStudyBarcoded
-                return ScRnaSeqStudyBarcodedSerializer(context=self.context).to_internal_value(data)
+                return ScRnaSeqStudyBarcodedSerializer(context=self.context)\
+                    .to_internal_value(data)
 
             else:
                 self.Meta.model = Study

--- a/hubmap/dataportal/tests.py
+++ b/hubmap/dataportal/tests.py
@@ -1,7 +1,7 @@
 from django.test import TestCase
 from rest_framework.test import APIClient
 
-from .models import *
+from .models import DataType, Institution, ScRnaSeqStudyCDNA, Study, Tissue
 from .views import StudyDetailView
 
 # TODO : complete tests

--- a/hubmap/dataportal/urls.py
+++ b/hubmap/dataportal/urls.py
@@ -1,4 +1,4 @@
-from django.conf.urls import *
+from django.conf.urls import url
 from django.urls import path
 
 from . import views

--- a/hubmap/dataportal/utils.py
+++ b/hubmap/dataportal/utils.py
@@ -2,9 +2,30 @@ from collections import OrderedDict
 from itertools import chain
 
 from django.db.models import Q
+from rest_framework import serializers
 
-from .models import *
-from .serializers import *
+from .models import (
+    Gene,
+    MassCytometryStudy,
+    MicroscopyStudy,
+    Protein,
+    ScAtacSeqStudy,
+    ScRnaSeqStudyBarcoded,
+    ScRnaSeqStudyCDNA,
+    SeqFishImagingStudy,
+    SpatialTranscriptomicStudy
+)
+from .serializers import (
+    GeneSerializer,
+    MassCytometryStudySerializer,
+    MicroscopyStudySerializer,
+    ProteinSerializer,
+    ScAtacSeqStudySerializer,
+    ScRnaSeqStudyBarcodedSerializer,
+    ScRnaSeqStudyCDNASerializer,
+    SeqFishImagingStudySerializer,
+    SpatialTranscriptomicStudySerializer
+)
 
 # In case we have to use recursive gene/gene expressions/new features, use this.
 # For now: keeping this.

--- a/hubmap/dataportal/utils.py
+++ b/hubmap/dataportal/utils.py
@@ -6,7 +6,8 @@ from django.db.models import Q
 from .models import *
 from .serializers import *
 
-'''in case we have to use recursive gene/gene expressions/new features, use this. for now: keeping this'''
+# In case we have to use recursive gene/gene expressions/new features, use this.
+# For now: keeping this.
 
 
 class RecursiveSerializer(serializers.Serializer):
@@ -45,29 +46,37 @@ def get_response_for_request(self, request, format=None):
         genes = get_genes_list(query)
         proteins = get_proteins_list(query)
         self.queryset = list(
-            chain(scrna_barcorded, scrna_cdna, scrna_atac, microscopy, mass_cytometry, spatial_transcriptomic, genes,
-                  proteins))
+            chain(scrna_barcorded, scrna_cdna, scrna_atac, microscopy,
+                  mass_cytometry, spatial_transcriptomic, genes, proteins))
     else:
         #first set query set
         self.queryset = list(
-            chain(scrna_barcorded, scrna_cdna, scrna_atac, microscopy, mass_cytometry, spatial_transcriptomic))
+            chain(scrna_barcorded, scrna_cdna, scrna_atac, microscopy,
+                  mass_cytometry, spatial_transcriptomic))
     self.queryset.sort(key=lambda x: x.id)
     #set context
     context = {
         "request": request,
     }
     #get serializers lists
-    scrna_barcorded_serializer = ScRnaSeqStudyBarcodedSerializer(scrna_barcorded, many=True, context=context)
-    scrna_cdna_serializer = ScRnaSeqStudyCDNASerializer(scrna_cdna, many=True, context=context)
-    spatial_transcriptomic_serializer = SpatialTranscriptomicStudySerializer(spatial_transcriptomic, many=True,
-                                                                             context=context)
-    microscopy_serializer = MicroscopyStudySerializer(microscopy, many=True, context=context)
-    mass_cytometry_serializer = MassCytometryStudySerializer(mass_cytometry, many=True, context=context)
-    scrna_atac_serializer = ScAtacSeqStudySerializer(scrna_atac, many=True, context=context)
-    seq_fish_imaging_serializer = SeqFishImagingStudySerializer(seq_fish_imaging, many=True, context=context)
-    response = scrna_barcorded_serializer.data + scrna_cdna_serializer.data + scrna_atac_serializer.data + \
-        microscopy_serializer.data + mass_cytometry_serializer.data + spatial_transcriptomic_serializer.data \
-        + seq_fish_imaging_serializer.data
+    scrna_barcorded_serializer = ScRnaSeqStudyBarcodedSerializer(
+        scrna_barcorded, many=True, context=context)
+    scrna_cdna_serializer = ScRnaSeqStudyCDNASerializer(
+        scrna_cdna, many=True, context=context)
+    spatial_transcriptomic_serializer = SpatialTranscriptomicStudySerializer(
+        spatial_transcriptomic, many=True, context=context)
+    microscopy_serializer = MicroscopyStudySerializer(
+        microscopy, many=True, context=context)
+    mass_cytometry_serializer = MassCytometryStudySerializer(
+        mass_cytometry, many=True, context=context)
+    scrna_atac_serializer = ScAtacSeqStudySerializer(
+        scrna_atac, many=True, context=context)
+    seq_fish_imaging_serializer = SeqFishImagingStudySerializer(
+        seq_fish_imaging, many=True, context=context)
+    response = (
+        scrna_barcorded_serializer.data + scrna_cdna_serializer.data
+        + scrna_atac_serializer.data + microscopy_serializer.data + mass_cytometry_serializer.data
+        + spatial_transcriptomic_serializer.data + seq_fish_imaging_serializer.data)
     response.sort(key=lambda x: x['id'])
     if query is not None:
         genes_serializer = GeneSerializer(genes, many=True, context=context)
@@ -171,7 +180,8 @@ def get_proteins_list(query):
     if query is None:
         proteins = Protein.objects.all()
     else:
-        proteins = Protein.objects.filter(Q(name__icontains=query) | Q(gene__hugo_symbol__icontains=query))
+        proteins = Protein.objects.filter(
+            Q(name__icontains=query) | Q(gene__hugo_symbol__icontains=query))
     return proteins
 
 

--- a/hubmap/dataportal/utils.py
+++ b/hubmap/dataportal/utils.py
@@ -55,7 +55,7 @@ def get_serializer_class(this_study):
 
 def get_response_for_request(self, request, format=None):
     query = self.request.query_params.get('search', None)
-    #get all individual study lists
+    # Get all individual study lists
     scrna_barcorded = get_scrna_barcorded_list(query)
     scrna_cdna = get_scrna_cdna_list(query)
     spatial_transcriptomic = get_spatial_transcriptomic_list(query)
@@ -70,16 +70,16 @@ def get_response_for_request(self, request, format=None):
             chain(scrna_barcorded, scrna_cdna, scrna_atac, microscopy,
                   mass_cytometry, spatial_transcriptomic, genes, proteins))
     else:
-        #first set query set
+        # First set query set
         self.queryset = list(
             chain(scrna_barcorded, scrna_cdna, scrna_atac, microscopy,
                   mass_cytometry, spatial_transcriptomic))
     self.queryset.sort(key=lambda x: x.id)
-    #set context
+    # Set context
     context = {
         "request": request,
     }
-    #get serializers lists
+    # Get serializers lists
     scrna_barcorded_serializer = ScRnaSeqStudyBarcodedSerializer(
         scrna_barcorded, many=True, context=context)
     scrna_cdna_serializer = ScRnaSeqStudyCDNASerializer(
@@ -186,13 +186,13 @@ def get_genes(self, request):
 
     genes = get_genes_list(query)
     self.queryset = genes
-    #set context
+    # Set context
     context = {
         "request": request,
     }
     print(genes)
     print(GeneSerializer(genes, many=True, context=context))
-    #get serializers lists
+    # Get serializers lists
     response = GeneSerializer(genes, many=True, context=context).data
     return response
 

--- a/hubmap/dataportal/views.py
+++ b/hubmap/dataportal/views.py
@@ -6,8 +6,19 @@ from rest_framework import generics, status, views
 from rest_framework.pagination import PageNumberPagination
 from rest_framework.response import Response
 
-from .serializers import *
-from .utils import *
+from .models import Study
+from .utils import (
+    DataType,
+    GeneSerializer,
+    Institution,
+    StudyListSerializer,
+    StudySerializer,
+    Tissue,
+    TissueColorSerializer,
+    get_genes,
+    get_response_for_request,
+    get_serializer_class
+)
 
 #TODO: Add OpenApi -> Swagger to rest framework
 #TODO: Add post request implementations

--- a/hubmap/dataportal/views.py
+++ b/hubmap/dataportal/views.py
@@ -20,8 +20,8 @@ from .utils import (
     get_serializer_class
 )
 
-#TODO: Add OpenApi -> Swagger to rest framework
-#TODO: Add post request implementations
+# TODO: Add OpenApi -> Swagger to rest framework
+# TODO: Add post request implementations
 
 
 class PaginationClass(PageNumberPagination):
@@ -51,7 +51,7 @@ class StudyListView(generics.GenericAPIView):
         print(serializer.data)
         return self.get_paginated_response(serializer.data)
 
-#TODO : deifne what fields are modifiable and what can be created
+    # TODO : define what fields are modifiable and what can be created
     def post(self, request, format=None):
         serializer = StudySerializer(data=request.data)
         if serializer.is_valid():

--- a/hubmap/dataportal/views.py
+++ b/hubmap/dataportal/views.py
@@ -6,18 +6,14 @@ from rest_framework import generics, status, views
 from rest_framework.pagination import PageNumberPagination
 from rest_framework.response import Response
 
-from .models import Study, DataType
-from .utils import (
+from .models import DataType, Institution, Study, Tissue
+from .serializers import (
     GeneSerializer,
-    Institution,
     StudyListSerializer,
     StudySerializer,
-    Tissue,
-    TissueColorSerializer,
-    get_genes,
-    get_response_for_request,
-    get_serializer_class
+    TissueColorSerializer
 )
+from .utils import get_genes, get_response_for_request, get_serializer_class
 
 # TODO: Add OpenApi -> Swagger to rest framework
 # TODO: Add post request implementations

--- a/hubmap/dataportal/views.py
+++ b/hubmap/dataportal/views.py
@@ -23,9 +23,13 @@ class StudyListView(generics.GenericAPIView):
 
     def get(self, request, format=None):
         response = get_response_for_request(self, request, format)
-        cell_count_summary = serialize_multi_dim_counts(compute_multi_dim_counts(self.queryset, "cell_count"))
-        image_count_summary = serialize_multi_dim_counts(compute_multi_dim_counts(self.queryset, "image_count"))
-        response.append({"summary": [{"cell_count": cell_count_summary}, {"image_count": image_count_summary}]})
+        cell_count_summary = serialize_multi_dim_counts(
+            compute_multi_dim_counts(self.queryset, "cell_count"))
+        image_count_summary = serialize_multi_dim_counts(
+            compute_multi_dim_counts(self.queryset, "image_count"))
+        response.append({"summary": [
+            {"cell_count": cell_count_summary}, {"image_count": image_count_summary}
+        ]})
         return Response(response)
 
     def list(self, request):

--- a/hubmap/dataportal/views.py
+++ b/hubmap/dataportal/views.py
@@ -6,9 +6,8 @@ from rest_framework import generics, status, views
 from rest_framework.pagination import PageNumberPagination
 from rest_framework.response import Response
 
-from .models import Study
+from .models import Study, DataType
 from .utils import (
-    DataType,
     GeneSerializer,
     Institution,
     StudyListSerializer,

--- a/hubmap/fixtures/add_tissuegene_heatmaps2db.py
+++ b/hubmap/fixtures/add_tissuegene_heatmaps2db.py
@@ -1,6 +1,6 @@
 import matplotlib.cm
 
-from dataportal.models import *
+from dataportal.models import Gene, TissueExpressionHeatmap
 
 
 def rgba_float_to_rgb_hex(floats):

--- a/hubmap/setup.cfg
+++ b/hubmap/setup.cfg
@@ -1,7 +1,6 @@
 [flake8]
 max-line-length = 99
 ignore =
-  E265, # TODO? "block comment should start with '# '"
   W503 # "line break before binary operator": Prefer operator at start of line so the context is clear.
 exclude =
   */migrations/*.py

--- a/hubmap/setup.cfg
+++ b/hubmap/setup.cfg
@@ -2,7 +2,6 @@
 max-line-length = 99
 ignore =
   E265, # TODO? "block comment should start with '# '"
-  F405, # TODO: 96 violations: "may be undefined, or defined from star imports"
   W503 # "line break before binary operator": Prefer operator at start of line so the context is clear.
 exclude =
   */migrations/*.py

--- a/hubmap/setup.cfg
+++ b/hubmap/setup.cfg
@@ -2,10 +2,11 @@
 max-line-length = 99
 ignore =
   E265, # TODO? "block comment should start with '# '"
-  E501, # TODO: Line length: add ignore for migrations
   F403, # TODO: * imports: "unable to detect undefined names", causes...
   F405, # TODO: 96 violations: "may be undefined, or defined from star imports"
   W503 # "line break before binary operator": Prefer operator at start of line so the context is clear.
+exclude =
+  */migrations/*.py
 
 [isort]
 line_length = 99

--- a/hubmap/setup.cfg
+++ b/hubmap/setup.cfg
@@ -2,7 +2,6 @@
 max-line-length = 99
 ignore =
   E265, # TODO? "block comment should start with '# '"
-  F403, # TODO: * imports: "unable to detect undefined names", causes...
   F405, # TODO: 96 violations: "may be undefined, or defined from star imports"
   W503 # "line break before binary operator": Prefer operator at start of line so the context is clear.
 exclude =

